imnodes.cpp

```diff
@@ -1035,36 +1034,11 @@ void ClickInteractionUpdate(ImNodesEditorContext& editor)
         }
     }
     break;
-    case ImNodesClickInteractionType_MiniMapPanning:
-    {
-        const bool dragging = GImNodes->AltMouseDragging;
-
-        if (dragging)
-        {
-            editor.Panning += ImGui::GetIO().MouseDelta / GImNodes->MiniMapZoom;
-        }
-        else
-        {
-            editor.ClickInteraction.Type = ImNodesClickInteractionType_None;
-        }
-    }
-    break;
-    case ImNodesClickInteractionType_MiniMapZooming:
-    {
-        GImNodes->MiniMapZoom = fmaxf(
-            0.05f,
-            fminf(
-                GImNodes->MiniMapZoom +
-                    0.1f * GImNodes->MiniMapZoom * GImNodes->AltMouseScrollDelta,
-                1.f));
-        editor.ClickInteraction.Type = ImNodesClickInteractionType_None;
-    }
-    break;
-    case ImNodesClickInteractionType_MiniMapSnapping:
+    case ImNodesClickInteractionType_CenterOnRequest:
     {
-        editor.Panning += GImNodes->MiniMapRectSnappingOffset;
-        GImNodes->MiniMapRectSnappingOffset = ImVec2(0.f, 0.f);
-
+        ImVec2 center = GImNodes->CanvasRectScreenSpace.GetSize() * 0.5f;
+        editor.Panning = ImVec2(0.0f, 0.0f) - editor.CenterOnRequest + center;
+        editor.CenterOnRequest = ImVec2(0.f, 0.f);
```

Comment:
Does this have to be handled in `ClickInteractionUpdate`? This updates the coordinate system, but doesn't have to be checked over multiple frames. Keeping this in `MiniMapUpdate` would make the calculation more localized.

Reply:
No, just thought that maybe this fits the existing architecture better. Removed it.